platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -236,6 +237,18 @@ <T extends SelfSerializable> void writeObjectByChildIndex(int index, List<T> lis
         }
     }
 
+    @Override
+    public void preHandle(
+            @NonNull Event event,
+            @NonNull Consumer<ScopedSystemTransaction<StateSignatureTransaction>> stateSignatureTransaction) {
+        ((ConsensusEvent) event).consensusTransactionIterator().forEachRemaining(transaction -> {
```

Comment:
Why cast this to a ConsensusEvent?

Reply:
`consensusTransactionIterator` method is not part of Event interface. It's part of ConsensusEvent specification

Reply:
But we are not getting ConsensusEvents in preHandle inside app. Good catch. I will adapt it.